chrome/content/zotero/integration/quickFormat.js

```diff
@@ -1296,54 +1426,107 @@ var Zotero_QuickFormat = new function () {
 		io.accept();
 	}
 	
+	// Flag any duplicate Enter or Escape events coming in within MILISECONDS_BETWEEN_KEYPRESSES
+	// of each other
+	this.throttleEscapeEnter = function (e) {
+		let now = new Date();
+		if (accepted) {
+			e.stopPropagation();
+			e.preventDefault();
+			return;
+		}
+		if (e.key == "Enter") {
+			if (lastEnter && now - lastEnter < MILISECONDS_BETWEEN_KEYPRESSES) {
+				throttleEnter = true;
+				return;
+			}
+			throttleEnter = false;
+			lastEnter = new Date();
+		}
+		else if (e.key == "Escape") {
+			if (lastEscape && now - lastEscape < MILISECONDS_BETWEEN_KEYPRESSES) {
+				throttleEscape = true;
+				return;
+			}
+			lastEscape = new Date();
+			throttleEscape = false;
+		}
+	};
+
 	/**
 	 * Handle escape for entire window
 	 */
-	this.onKeyPress = function (event) {
+	this.onWindowKeyPress = function (event) {
+		if (accepted) return;
 		var keyCode = event.keyCode;
-		if (keyCode === event.DOM_VK_ESCAPE && !accepted) {
+		if (keyCode === event.DOM_VK_ESCAPE && !throttleEscape) {
 			accepted = true;
 			io.citation.citationItems = [];
 			io.accept();
 			window.close();
 		}
+		else if (event.key == "Enter" && !throttleEnter) {
+			event.preventDefault();
+			Zotero_QuickFormat._accept();
+		}
 	};
 
+	function moveFocusForward(node) {
+		if (node.nextElementSibling?.focus) {
+			node.nextElementSibling.focus();
+			return true;
+		}
+		return false;
+	}
+
+	function moveFocusBack(node) {
+		// Skip line break if it's before the node
+		if (node.previousElementSibling?.tagName == "BR") {
```

Comment:
Those line breaks are not strictly needed but it's an addition that I think slightly improves the experience. 

Suppose the dialog has 2 lines with 6 bubbles - 4 on the 1st line and 2 on the second. One clicks right before the first bubble on the second line (bubble 5), `getLastBubbleBeforePoint` will tell us that the placeholder should be inserted after bubble 4. When the input is added, it can end up going to the very end of line 1, as opposed to the start of line 2, if there's space there. I thought it's a bit odd because you click in one location and the input shows up in another - if we add a `<br>` before the input, it makes sure that it ends up on the second line.

So this is not necessary, I just thought it's a bit nicer that way